datasync/chaindatafetcher/kas/contract_caller.go

```diff
@@ -0,0 +1,192 @@
+}
+
+func (f *contractCaller) CallContract(ctx context.Context, call klaytn.CallMsg, blockNumber *big.Int) ([]byte, error) {
+	callArgs := api.CallArgs{
+		From: call.From,
+		To:   call.To,
+		Data: hexutil.Bytes(call.Data),
+	}
+	return f.blockchainAPI.Call(ctx, callArgs, rpc.BlockNumber(blockNumber.Int64()))
+}
+
+func getCallOpts(blockNumber *big.Int, timeout time.Duration) (*bind.CallOpts, context.CancelFunc) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	return &bind.CallOpts{Context: ctx, BlockNumber: blockNumber}, cancel
+}
+
+// the `SupportsInterface` method error must be handled with the following cases.
+// case 1: the contract implements fallback function
+// - the call can be reverted within fallback function: returns "evm: execution reverted"
+// - the call can be done successfully, but it outputs empty: returns "abi: unmarshalling empty output"
+// case 2: the contract does not implements fallback function
+// - the call can be reverted: returns "evm: execution reverted"
+// handleSupportsInterfaceErr handles the given error according to the above explanation.
+func handleSupportsInterfaceErr(err error) error {
+	if err != nil && (strings.Contains(err.Error(), errMsgEmptyOutput) || strings.Contains(err.Error(), errMsgEvmReverted)) {
+		return nil
+	}
+	return err
+}
+
+// supportsInterface returns true if the given interfaceID is supported, otherwise returns false.
+func (f *contractCaller) supportsInterface(contract common.Address, opts *bind.CallOpts, interfaceID [4]byte) (bool, error) {
+	caller, err := kip13.NewInterfaceIdentifierCaller(contract, f)
+	if err != nil {
+		logger.Error("NewInterfaceIdentifierCaller is failed", "contract", contract.String(), "interfaceID", hexutil.Encode(interfaceID[:]))
+		return false, err
+	}
+	isSupported, err := caller.SupportsInterface(opts, interfaceID)
+	return isSupported, handleSupportsInterfaceErr(err)
+}
+
+// isKIP13 checks if the given contract implements KIP13 interface or not at the given block.
+func (f *contractCaller) isKIP13(contract common.Address, blockNumber *big.Int) (bool, error) {
+	var (
+		opts   *bind.CallOpts
+		cancel context.CancelFunc
+	)
+	opts, cancel = getCallOpts(blockNumber, f.callTimeout)
+	defer cancel()
+	if isKIP13, err := f.supportsInterface(contract, opts, IKIP13Id); err != nil {
+		logger.Error("supportsInterface is failed", "contract", contract.String(), "blockNumber", blockNumber, "interfaceID", hexutil.Encode(IKIP13Id[:]))
+		return false, err
+	} else if !isKIP13 {
+		return false, nil
+	}
+
+	opts, cancel = getCallOpts(blockNumber, f.callTimeout)
+	defer cancel()
+	if isInvalid, err := f.supportsInterface(contract, opts, InvalidId); err != nil {
+		logger.Error("supportsInterface is failed", "contract", contract.String(), "blockNumber", blockNumber, "interfaceID", hexutil.Encode(InvalidId[:]))
+		return false, err
+	} else if isInvalid {
+		return false, nil
+	}
```

Comment:
Is checking `InvalidId ` mandatory?  If it is, why don't you check `InvalidId` in `isKIP7`?